### ISSUE TITLE
Microsoft's Envelope maker does not read with NVDA

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#480]
**Reported by scottbb on 2009-11-25 20:08**
When you go to Tools/Letters/Enveliopes you do not get very good feedback when you tab around in this dialog box.  this should be addressed for the next release because so many people use Microsoft Office Word.

**Comment by leonardder** [comment]
Since @josephsl didn't get any comment on his request from like three years ago, how about a won't fix?

**Comment by PratikP1** [comment]
I disagree with this recommendation. I thought there was another ticket regarding this. But, the envelope creator/designer does not work in current office releases. 

Tested in Office 2016. One of the problems it suffers from is similar to Outlook's signature creator.

**Comment by jcsteh** [comment]
[**@PratikP1**](https://github.com/PratikP1) commented on [9 Jun 2017, 07:27 GMT+10](https://github.com/nvaccess/nvda/issues/480#issuecomment-307232399 "2017-06-08T21:27:34Z - Replied by Github Reply Comments"):
> Tested in Office 2016. One of the problems it suffers from is similar to Outlook's signature creator.

In that case, it might be fixed or improved by #7254 (currently in code review).

**Comment by PratikP1** [comment]
I’ll report back once #7254<https://github.com/nvaccess/nvda/pull/7254> is incubated.

**Comment by Qchristensen** [comment]
I just had a look at this with Word 2016.

Using NVDA 2017.2, nothing is announced when I open the dialog; the focus is in the address edit.  When I tab around, I can read the rest of the dialog.  Entering text in the address field, I don't get any information other than typing echo.  If I navigate around text in the edit there is no feedback.

Using NVDA Next snapshot 14146,35ffacb3, I can interact with the edit field as expected.  NVDA announces it as a "document" rather than an edit, but everything seems to work.

**Comment by PratikP1** [comment]
As I suspected, now that #7254 is incubating, most of the problems with the envelope maker are gone. I can confirm @qchristensen's findings. Other than "edit" not being announced, this edit field also does not announce characters being deleted with back space. Outlook's signature editor calls this kind of field "edit document." In all other respects, fields in Word's envelope maker work now.

**Comment by ehollig** [comment]
@PratikP1, would you say that this issue can now be closed?

**Comment by PratikP1** [comment]
@ehollig Sorry it took me so long to get back to you on this. While I wouldn't describe the user interaction to be the best in the envelope maker, at least, it now speaks all the elements. The text edit fields work as expected. There is no label applied to the two address fields. But alt+d and alt+r move the user to the Address and Return address fields respectively. Unlike other selection commands, select all does not work. In other words, there are a few expectations defied by the document/edit classification of the two respective text fields. The usability could be improved a bit.

**Comment by Adriani90** [comment]
@PratikP1 thank you for reporting back. I will close this issue for now. Please open new issues for every specific behavior which is not as expected. This will make it easier to see what works and what is still missing. Thanks.
